### PR TITLE
Implemented LOOP, LEAVE, and ITERATE, partially implemented DECLARE ... HANDLER

### PR DESCRIPTION
**Pull request by Hydrocharged**
This needs many, many, many more tests. Also should rewrite how stored procedures are analyzed as well, but that's a larger project overall.

### CODE REVIEW

File: sql/analyzer/declare.go [hunk 1]
```diff
@@ -48,30 +49,41 @@ type declarationCursor struct {
...
 	nextVarId  *int
 	nextCurId  *int
+	nextHandId *int
 	parent     *declarationScope
 	conditions map[string]*plan.DeclareCondition
 	variables  map[string]*declarationVariable
 	cursors    map[string]*declarationCursor
+	loops      map[string]struct{}
+
+	//TODO: implement proper handler support
```
Comment (zachmu) [inline sql/analyzer/declare.go:59]: Can we get at least one skipped test where there are TODOs for functionality?

File: sql/parse/parse.go [hunk 2]
```diff
@@ -1360,6 +1372,31 @@ func convertDeclareCursor(ctx *sql.Context, d *sqlparser.Declare) (sql.Node, err
...
+	var action plan.DeclareHandlerAction
+	switch dHandler.Action {
+	case sqlparser.DeclareHandlerAction_Continue:
+		action = plan.DeclareHandlerAction_Continue
+	case sqlparser.DeclareHandlerAction_Exit:
+		action = plan.DeclareHandlerAction_Exit
+	case sqlparser.DeclareHandlerAction_Undo:
+		action = plan.DeclareHandlerAction_Undo
+	default:
+		return nil, fmt.Errorf("unknown DECLARE ... HANDLER action: %v", dHandler.Action)
```
Comment (zachmu) [inline sql/parse/parse.go:1395]: Shouldn't this be unsupported feature err?

File: sql/parse/parse.go [hunk 3]
```diff
@@ -1372,6 +1409,22 @@ func convertClose(ctx *sql.Context, closeCursor *sqlparser.CloseCursor) (sql.Nod
 	return plan.NewClose(closeCursor.Name), nil
 }
 
+func convertLoop(ctx *sql.Context, loop *sqlparser.Loop, query string) (sql.Node, error) {
```
Comment (zachmu) [inline sql/parse/parse.go:1412]: Parse tests for these?

Comment (zachmu) [reply sql/parse/parse.go:1412]: Just one or two parse tests for all the features in stored procs is probably plenty

**Review by zachmu — APPROVED** [review]
LGTM, just a few comments

File: sql/analyzer/declare.go (hunk 1 above)
Comment (Hydrocharged) [reply sql/analyzer/declare.go:59]: I'mma work on completing stored procedures soon, so no point for now.

File: sql/parse/parse.go (hunk 2 above)
Comment (Hydrocharged) [reply sql/parse/parse.go:1395]: Nope, as there should only be the three listed. If there's another one, then there shouldn't be and it's unknown.

**Pull request merged**